File: newtons_method.py
from sympy import *
init_printing(use_unicode=True)
x=symbols('x')



def newtons_method(expr,epsilon,x_start):
    x=symbols('x')
    f=sympify(expr)
    x_naught=float(x_start)
    df=diff(f,x)

    print"\n\n"
    print'#           x'
    print'0        ', x_naught

    for i in range(1,200):   
    
        f_naught=f.subs(x,x_naught)
        df_naught=df.subs(x,x_naught)
  
        x_one=x_naught-(f_naught/df_naught)   
  
  
        if abs(x_one-x_naught) < epsilon:
            break
          
  
        print i,"        ",x_one
  
        x_naught=x_one



function=str(input("Provide a function: "))
ep=float(input("Provide an epsilon: "))
initialx=float(input("And give an initial guess: "))
newtons_method(function,ep,initialx)

File: newtons_method.m
clear
format long
syms x;



%CHANGE THESE VALUES FOR THE PROBLEM:
epsilon=.1;
x0=-.7;
f=x^3+x-1;


fprintf("\n\n")
fprintf("#       x\n")
fprintf("0    %.8f\n",x0)
df=diff(f,x);

for i=1:200
   
    
  f0=subs(f,x,x0);
  df0=subs(df,x,x0);
  
  x1=x0-f0/df0;
   
  
  
  if abs(x1-x0) < epsilon
      break
  end  
  
 fprintf("%d    %.8f\n",i,x1)
  
  x0=x1;
  
end  